### PR TITLE
Implement viewport scrolling commands

### PR DESCRIPTION
**Pull request by tremby**
This is for #65.

I've become stuck. @ambientsound, any chance you can take a look and help me figure out why this isn't working?

### CODE REVIEW

**Comment by ambientsound** [comment]
The problem here is that the `views.ViewPort` object is passed by value, and so the connection to the original object is lost. By returning a pointer in `GetViewport`, this problem is solved.

However, I would strongly prefer that the viewport object is kept private, and that a `Scroll` method (or similar) is exposed instead. Please see the upstream [feature/viewport-commands](/ambientsound/pms/tree/feature/viewport-commands).

**Comment by tremby** [comment]
Nice! OK, I'll finish this off soon.

If you have a moment, can you show me how I would have fixed my own code?

(I understand that it's better to keep it private instead, I just want to explicitly see what I was doing wrong.)

File: api/interface.go [hunk 1]
```diff
@@ -2,11 +2,14 @@ package api
...
 import (
 	"github.com/ambientsound/pms/songlist"
+
+	"github.com/gdamore/tcell/views"
 )
 
 type SonglistWidget interface {
 	AddSonglist(songlist.Songlist)
 	FallbackSonglist() songlist.Songlist
+	GetViewport() views.ViewPort
```
Comment (ambientsound) [inline api/interface.go]: In order for this to work you would write:

```
GetViewport() *views.ViewPort
```

Comment (tremby) [reply api/interface.go]: Oh wow, it's as simple as that. Thanks.

**Comment by tremby** [comment]
Ready for review!

Thanks for the help.

**Comment by tremby** [comment]
I believe I have this working correctly both with `'center'` on and off, by the way. It works just like vim (with and without `'scrolloff'=999` in all cases I've tried) other than at the very bottom (vim lets you keep scrolling and gives you `~` lines).

**Pull request merged**